_rules/table-header-cells-refer-to-data-cells.md

```diff
@@ -0,0 +1,160 @@
+---
+name: All table header cells refer to data cells
+
+test_type: atomic
+
+description: |
+  This rule checks that each table header in a data table refers to data cells.
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Test Procedure
+
+### Applicability
+
+This rule applies to any element with [semantic role](#semantic-role) of `columnheader` or `rowheader` within the `table` element that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree).
```

Comment:
If you're talking about `table` elements, this should be limited to `th` elements IMO. I also think you should ignore any th that is not in the accessibility tree.

Another suggestion here would be to limit this rule to `th` elements in a table that has non-empty `td` elements: https://github.com/dequelabs/axe-core/issues/741#issuecomment-367957486